Build fails

1. Clone the repo
2. `npm install`
3. `npm run build-dev`

The build fails with error:

```
> broken-bounty@1.0.1 build-dev /home/user/Desktop/broken-bounty
> gulp clear-build && gulp dev-tasks && gulp embed && gulp html-dev

fs.js:41
} = primordials;
    ^

ReferenceError: primordials is not defined
    at fs.js:41:5
    at req_ (/home/user/Desktop/broken-bounty/node_modules/natives/index.js:143:24)
    at Object.req [as require] (/home/user/Desktop/broken-bounty/node_modules/natives/index.js:55:10)
    at Object.<anonymous> (/home/user/Desktop/broken-bounty/node_modules/graceful-fs/fs.js:1:37)
    at Module._compile (internal/modules/cjs/loader.js:1076:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
    at Module.load (internal/modules/cjs/loader.js:941:32)
    at Function.Module._load (internal/modules/cjs/loader.js:782:14)
    at Module.require (internal/modules/cjs/loader.js:965:19)
    at require (internal/modules/cjs/helpers.js:88:18)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! broken-bounty@1.0.1 build-dev: `gulp clear-build && gulp dev-tasks && gulp embed && gulp html-dev`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the broken-bounty@1.0.1 build-dev script.
```

Dependencies may need to be updated.

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -14,81 +14,7 @@ const del = require('del');
 // - run an html task
 
 //////////
-
-gulp.task('clear-build', ['delete-build'], function() {
-	// fs.mkdir('./dist', function(exception) {
-	// 	cb(exception);
-	// });
-
-	return Promise.all([
-		new Promise(function(resolve, reject) {
-			fs.mkdir('./dist', function(err) {
-				if (err) {
-					reject(err);
-				}
-				else {
-					resolve();
-				}
-			});
-		}),
-		new Promise(function(resolve, reject) {
-			fs.mkdir('./output', function(err) {
-				if (err) {
-					reject(err);
-				}
-				else {
-					resolve();
-				}
-			});
-		})
-	]);
-});
-
-gulp.task('dev-tasks',        ['css-dev',        'js-dev']);
-gulp.task('production-tasks', ['css-production', 'js-production']);
-
-gulp.task('embed', function() {
-	return Promise.all([
-		new Promise(function(resolve, reject) {
-			fs.readFile('./src/index.html', 'utf-8', function(err, html) {
-				if (err) reject(err);
-
-				resolve(html);
-			});
-		}),
-		new Promise(function(resolve, reject) {
-			fs.readFile('./output/style.css', 'utf-8', function(err, css) {
-				if (err) reject(err);
-
-				resolve(css);
-			});
-		}),
-		new Promise(function(resolve, reject) {
-			fs.readFile('./output/bundle.js', 'utf-8', function(err, js) {
-				if (err) reject(err);
-
-				resolve(js);
-			});
-		})
-	]).then(function(values) {
-		var html = values[0];
-		var css = values[1];
-		var js = values[2];
-
-		html = html.replace('/* css hook */', css);
-		html = html.replace('/* js hook */', js);
-
-		return new Promise(function(resolve, reject) {
-			fs.writeFile('./output/index.html', html, function(err) {
-				if (err) reject(err);
-
-				resolve();
-			});
-		});
-	});
-});
-
-//////////
+// Task Define
 
 gulp.task('delete-build', function() {
 	// return del(['./dist']);
@@ -184,6 +110,82 @@ gulp.task('html-production', function() {
 
 		return new Promise(function(resolve, reject) {
 			fs.writeFile('./dist/index.html', result, function(err) {
+				if (err) reject(err);
+
+				resolve();
+			});
+		});
+	});
+});
+
+//////////
+// Task Execute
+gulp.task('clear-build', gulp.series('delete-build', function(){
+	// fs.mkdir('./dist', function(exception) {
+	// 	cb(exception);
+	// });
+
+	return Promise.all([
+		new Promise(function(resolve, reject) {
+			fs.mkdir('./dist', function(err) {
+				if (err) {
+					reject(err);
+				}
+				else {
+					resolve();
+				}
+			});
+		}),
+		new Promise(function(resolve, reject) {
+			fs.mkdir('./output', function(err) {
+				if (err) {
+					reject(err);
+				}
+				else {
+					resolve();
+				}
+			});
+		})
+	]);
+}));
+
+gulp.task('dev-tasks', gulp.series('css-dev', 'js-dev'));
+gulp.task('production-tasls', gulp.series('css-production', 'js-production'));
+
+
+gulp.task('embed', function() {
+	return Promise.all([
+		new Promise(function(resolve, reject) {
+			fs.readFile('./src/index.html', 'utf-8', function(err, html) {
+				if (err) reject(err);
+
+				resolve(html);
+			});
+		}),
+		new Promise(function(resolve, reject) {
+			fs.readFile('./output/style.css', 'utf-8', function(err, css) {
+				if (err) reject(err);
+
+				resolve(css);
+			});
+		}),
+		new Promise(function(resolve, reject) {
+			fs.readFile('./output/bundle.js', 'utf-8', function(err, js) {
+				if (err) reject(err);
+
+				resolve(js);
+			});
+		})
+	]).then(function(values) {
+		var html = values[0];
+		var css = values[1];
+		var js = values[2];
+
+		html = html.replace('/* css hook */', css);
+		html = html.replace('/* js hook */', js);
+
+		return new Promise(function(resolve, reject) {
+			fs.writeFile('./output/index.html', html, function(err) {
 				if (err) reject(err);
 
 				resolve();

--- a/package.json
+++ b/package.json
@@ -13,7 +13,7 @@
     "babel-preset-es2015": "6.13.2",
     "cssnano": "3.7.3",
     "del": "2.2.2",
-    "gulp": "3.9.1",
+    "gulp": "4.0.2",
     "html-minifier": "3.0.2",
     "postcss": "5.1.2",
     "postcss-advanced-variables": "1.2.2",

